core/services/head_tracker_test.go

```diff
@@ -254,7 +256,12 @@ func TestHeadTracker_StartConnectsFromLastSavedHeader(t *testing.T) {
 	t.Parallel()
 	g := gomega.NewGomegaWithT(t)
 
-	store, cleanup := cltest.NewStore(t)
+	// Need separate db because ht.Stop() will cancel the ctx, causing a db connection
```

Comment:
Urgh. Do you think we should use channels instead of contexts for closing? Would that help?

Reply:
I thought a bit more about the support we have for cancelling in flight queries and I think it is desirable. We don't want some hanging db op stop the head tracker from closing. I don't think we should change our code just to cater to the go-txdb testing lib, so this seemed like the best approach